### PR TITLE
Complete tests for `lib/record.js`

### PR DESCRIPTION
**Pull request by rmeritz**

### CODE REVIEW

File: test/delete.test.js [hunk 1]
```diff
@@ -40,6 +40,29 @@ describe('record deletion', function() {
...
+            });
+        });
+
+        return airtable
+            .base('app123')
+            .table('Table')
+            .destroy('rec123')
+            .then(
+                function() {
+                    throw new Error('Promise unexpectly fufilled.');
```
Comment (jugglinmike) [inline test/delete.test.js:56]: Typo: "unexpectly" -> "unexpectedly" (here and below)

File: test/delete.test.js [hunk 2]
```diff
@@ -52,7 +75,7 @@ describe('record deletion', function() {
             });
     });
 
-    it('can throw an error if delete fails', function(done) {
+    it('can throw an error if a multi-record delete fails', function(done) {
```
Comment (jugglinmike) [inline test/delete.test.js:78]: Good catch

File: test/record.test.js [hunk 3]
```diff
@@ -170,6 +190,30 @@ describe('Record', function() {
...
+            record.set('foo', undefined); // eslint-disable-line no-undefined
+            record.set('baz', 'qux');
+            record.save(function(err, updatedRecord) {
+                expect(err).toBeNull();
+
+                expect(updatedRecord).toBe(record);
+                expect(record.get('foo')).toBeUndefined();
+                expect(record.get('baz')).toEqual('qux');
+
+                expect(table._base.runAction).toHaveBeenCalledWith(
```
Comment (jugglinmike) [inline test/record.test.js:203]: The invocation of `_base.runAction` doesn't seem particularly important for the caller. A more robust test would involve verifying that an HTTP PUT request is issued. This file sets the precedent, so good on you for finding that and sticking to it! Improving the existing tests is in-scope for our work, but I'll be satisfied if you'd rather address this in a follow-up patch.

File: test/delete.test.js [hunk 4]
```diff
@@ -40,6 +40,29 @@ describe('record deletion', function() {
...
+            .table('Table')
+            .destroy('rec123')
+            .then(
+                function() {
+                    throw new Error('Promise unexpectly fufilled.');
+                },
+                function(err) {
+                    expect(err.statusCode).toBe(402);
+                    expect(err.message).toBe('foo bar');
+                    done();
```
Comment (jugglinmike) [inline test/delete.test.js:61]: The `done` callback can be omitted when tests return a Promise (here and below)

File: test/delete.test.js (hunk 1 above)
Comment (rmeritz) [reply test/delete.test.js]: Done.

File: test/delete.test.js (hunk 4 above)
Comment (rmeritz) [reply test/delete.test.js]: Done!

File: test/record.test.js (hunk 3 above)
Comment (rmeritz) [reply test/record.test.js]: I updated these tests to use the same mocking we use in all other places.

**Comment by kasrak** [comment]
@jugglinmike any other comments?

File: test/delete.test.js (hunk 4 above)
Comment (jugglinmike) [reply test/delete.test.js]: Digging the pun

**Comment by jugglinmike** [comment]
None--this looks good to me!

**Pull request merged**